src/btreemap.rs

```diff
@@ -934,17 +965,30 @@ where
 
     // Saves the map to memory.
     fn save(&self) {
-        let header = BTreeHeader {
+        let header = BTreeHeaderV1 {
             magic: *MAGIC,
             version: LAYOUT_VERSION,
-            root_addr: self.root_addr,
             max_key_size: self.max_key_size,
             max_value_size: self.max_value_size,
+            root_addr: self.root_addr,
             length: self.length,
-            _buffer: [0; 24],
         };
 
-        write_struct(&header, Address::from(0), self.memory());
+        Self::write_header(&header, self.memory()).unwrap();
+    }
+
+    /// Write the layout header to the memory.
+    fn write_header(header: &BTreeHeaderV1, memory: &M) -> Result<(), crate::GrowFailed> {
+        // Serialize the header
+        let mut buf = [0; core::mem::size_of::<BTreeHeaderV1>()];
+        buf[0..3].copy_from_slice(&header.magic);
+        buf[3] = header.version;
+        buf[4..8].copy_from_slice(&header.max_key_size.to_le_bytes());
+        buf[8..12].copy_from_slice(&header.max_value_size.to_le_bytes());
+        buf[12..20].copy_from_slice(&header.root_addr.get().to_le_bytes());
+        buf[20..28].copy_from_slice(&header.length.to_le_bytes());
+        // Write the header
+        crate::safe_write(memory, 0, &buf)
```

Comment:
Since we're simply unwrapping the value returned here, we could use the `write` method that panics when growing fails.

```suggestion
        crate::write(memory, 0, &buf)
```